kpi-tracker/src/main.rs

```diff
@@ -22,56 +26,105 @@ struct Args {
         help = "The endpoint is expected to point to a concordium node grpc v2 API.",
         default_value = "http://localhost:20001"
     )]
-    node: Endpoint,
+    node:       Endpoint,
     /// How many blocks to process.
     // Only here for testing purposes...
     #[arg(long = "num-blocks", default_value_t = 10000)]
     num_blocks: u64,
 }
 
-/// Holds information about accounts created on chain.
+/// Information about individual blocks. Useful for linking entities to a block
+/// and it's corresponding attributes.
+#[derive(Debug, Clone, Copy)]
+struct BlockDetails {
+    /// Finalization time of the block. Used to show how metrics evolve over
+    /// time by linking entities, such as accounts and transactions, to
+    /// the block in which they are created.
+    block_time: DateTime<Utc>,
+    /// Height of block from genesis. Used to restart the process of collecting
+    /// metrics from the latest block recorded.
+    height:     AbsoluteBlockHeight,
+}
+
+/// Holds selected attributes about accounts created on chain.
 #[derive(Debug)]
 struct AccountDetails {
     /// Whether an account was created as an initial account or not.
     is_initial: bool,
-    /// Link to the block in which the account was created.
+    /// Foreign key to the block in which the account was created.
     block_hash: BlockHash,
 }
 
-/// Information about individual blocks. Useful for linking entities to a block and it's
-/// corresponding attributes.
-#[derive(Debug, Clone, Copy)]
-struct BlockDetails {
-    /// Finalization time of the block. Used to show how metrics evolve over time by linking entities, such as accounts and transactions, to
-    /// the block in which they are created.
-    block_time: DateTime<Utc>,
-    /// Height of block from genesis. Used to restart the process of collecting metrics from the
-    /// latest block recorded.
-    height: AbsoluteBlockHeight,
+/// Holds selected attributes of an account transaction.
+#[derive(Debug)]
+struct TransactionDetails {
+    /// The transaction type of the account transaction. Can be none if
+    /// transaction was rejected due to serialization failure.
+    transaction_type: Option<TransactionType>,
+    /// Foreign key to the block in which the transaction was finalized.
+    block_hash:       BlockHash,
+    /// The cost of the transaction.
+    cost:             Amount,
+    /// Whether the transaction failed or not.
+    is_failed:        bool,
```

Comment:
I would probably have a `is_succes` field instead. I find it weird to focus on the negative.